### DISCUSSION TITLE
TTS does not work properly on iOS/iPadOS

### DISCUSSION THREAD

**Discussion by Jeremy-Developer-Page** [open-webui/open-webui#5286] [General]
# Bug Report
If you refer to the TTS on the device this works correctly on Windows and macOS but if on iOS and iPadOS 14 to 17 (stable) and 18 (up to the last beta currently available) you make the list of voices for example in Italian the correct voices are not listed but only some random ones and regardless of which one you go to select the default one of the main language in use will always be used. This is a problem because on other sites this doesn't happen and if the entry is in low quality (as most basic entries are) you can't understand anything. I would therefore invite testing and investigation.

Translated with DeepL.com (free version)
## Installation Method
Docker Open WebUI/Ollama

## Environment
Open WebUI 0.3.21
Ollama 0.3.9

Linux (OpenWebUI and Ollama)
Browser(MS Edge Windows, iOS, iPadOS, macOS, Safari for iOS, iPadOS, macOS)

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
It is expected that the list will correctly list all the default language entries on the device, and selecting one will correctly set it up
## Actual Behavior:
The list includes random English language voices and sounds, not Italian, French, Spanish, etc. (even if the device is set to that language). Regardless of which voice you select it will keep the basic default one of the language being used by the device (for example for the Italian voice it will be Alice)

## Reproduction Details
Set the device language to another language, go to administrator panel settings and select Web entries for TTS. You will find that the list of items listed is not the same with the one shown in Phone Settings/Accessibility/Content Read Aloud/Voices/Selected Language (e.g., Italian). This is true for iOS and iPadOS

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots/Screen Recordings (if applicable):**
![IMG_0199](https://github.com/user-attachments/assets/f79a3e21-cb44-4b69-8979-facf46e58770)
![IMG_0200](https://github.com/user-attachments/assets/7fefff91-8727-46ea-a5a6-7552931e48bf)
![IMG_0201](https://github.com/user-attachments/assets/b53d0319-7728-4c2e-8a9b-1a67129b61c2)
![IMG_0202](https://github.com/user-attachments/assets/2497b339-dea6-46a3-b0a3-6c329ccc35be)
![IMG_0203](https://github.com/user-attachments/assets/530370fb-4074-42a4-8d78-66c2b8cd0bf1)
![IMG_0204](https://github.com/user-attachments/assets/71a8e2cf-9ecc-4685-8d85-59aefa9ba826)
![IMG_0196](https://github.com/user-attachments/assets/64918cb4-27e4-4d65-801b-77d4e2929cd3)
![IMG_0197](https://github.com/user-attachments/assets/15663346-8e58-4b54-b155-95c540016176)
![IMG_0198](https://github.com/user-attachments/assets/5a9635df-17c1-4509-8b88-62feda8617f6)

## Note

If you need further information or help with the resolution or tests do not hesitate to contact me, I am willing to help you solve a problem that creates serious inconvenience for those who use the TTS on a daily basis and can benefit from this fix. I also want to highlight that haptic feedback does not work on iOS, so if you could solve that problem it would also be perfect. Thank you very much and have a good day

**Comment by mihkhub** [comment]
the “Read Aloud” feature of OpenWebUI on iPhone or iOS.
